h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -210,18 +210,25 @@ public void onBuildFailure(final ParallelModelBuilder.ModelBuildFailure modelBui
 
     private void attemptBuildNextModel(final ParallelModelBuilder parallelModelBuilder, final Model previousModel) {
       // Attempt to train next model
-      final MP nextModelParams = getNextModelParams(hyperspaceIterator, previousModel, grid);
-      if (nextModelParams != null
-              && isThereEnoughTime()
-              && !_job.stop_requested()
-              && !_hyperSpaceWalker.stopEarly(previousModel, grid.getScoringInfos())) {
+      try {
+        // I hereby claim that the locking system below has been researched by Andrey Spiridonov
```

Comment:
when giving credits aways add @githubhandle or public linkedin profile name

Reply:
I apologize, I did not know that. I'll correct it asap.

Reply:
The credits were removed, as instructed by the author.